src/compiler/utilities.ts

```diff
@@ -11360,18 +11360,23 @@ export function createNameResolver({
                         location = root.parent;
                     }
                     break;
-                case SyntaxKind.Parameter:
+                case SyntaxKind.Parameter: {
+                    const param = location as ParameterDeclaration;

```

Comment:
style question: is there a reason for all the casts in this code? Maybe it's better to drop this alias and use `location as ParameterDeclaration)` everywhere.

Reply:
No specific reason, I just prefer to have a single cast than multiple ones that are the same for the same variable. I can change this if you prefer the other style.